src/main/java/org/broadinstitute/hellbender/utils/bwa/BwaMemAlignmentUtils.java

```diff
@@ -57,9 +67,15 @@ public static SAMRecord applyAlignment( final String readName, final byte[] base
                         }
                     }
                 }
-                bases = Arrays.copyOfRange(bases, alignment.getSeqStart(), alignment.getSeqEnd());
+            }
+            if (tmpCigar.getFirstCigarElement().getOperator() == CigarOperator.H ||
+                    tmpCigar.getLastCigarElement().getOperator() == CigarOperator.H) {
+                final int leftHardClip = tmpCigar.getFirstCigarElement().getOperator() == CigarOperator.H ? tmpCigar.getFirstCigarElement().getLength() : 0;
```

Comment:
The start and end of the alignment always exclude clipping, as far as I know.  Kind of by definition, right?